aptos-move/move-examples/staking/sources/commission.move

```diff
@@ -0,0 +1,119 @@
+        let now_secs = timestamp::now_seconds();
+        let seconds_passed = now_secs - config.last_update_secs;
+        let commission_earned = math64::mul_div(seconds_passed, config.yearly_commission_amount, ONE_YEAR_IN_SECONDS);
+
+        commission_earned + config.commission_debt
+    }
+
+    /// Can be called by the manager to change the yearly commission amount.
+    public entry fun set_yearly_commission_amount(manager: &signer, new_commission: u64) acquires CommissionConfig {
+        let config = &mut CommissionConfig[@staking];
+        assert!(signer::address_of(manager) == config.manager, EUNAUTHORIZED);
+        config.yearly_commission_amount = new_commission;
+    }
+
+    /// Can be called by the manager or operator to change the account that receives the commission.
+    public entry fun set_operator(account: &signer, new_operator: address) acquires CommissionConfig {
+        let config = &mut CommissionConfig[@staking];
+        let account_addr = signer::address_of(account);
+        assert!(account_addr == config.manager || account_addr == config.operator, EUNAUTHORIZED);
+        config.operator = new_operator;
+    }
+
+    /// Distribute the commission to operator and remaining amount to manager.
+    /// Can only be called by the manager or operator.
+    public entry fun distribute_commission(account: &signer) acquires CommissionConfig {
+        // Commission owed so far plus any debt.
+        let commission_owed = (commission_owed() as u128);
+
+        // Only manager or operator can call this function.
+        let config = &mut CommissionConfig[@staking];
+        let account_addr = signer::address_of(account);
+        assert!(account_addr == config.manager || account_addr == config.operator, EUNAUTHORIZED);
+        config.last_update_secs = timestamp::now_seconds();
+        config.commission_debt = 0;
+
+        // Commission in APT = commission earned / APT price.
+        let apt_price = oracle::get_apt_price();
+        let commission_in_apt = (math128::mul_div(commission_owed, apt_price, oracle::precision()) as u64);
+
+        let commission_signer = &account::create_signer_with_capability(&config.signer_cap);
+        let balance = coin::balance<AptosCoin>(@staking);
+        // If there's not enough balance to pay the commission, either commission rate is set too high or APT price is low.
+        // Otherwise, pay the operator the commission in APT and send remaining balance to the manager.
+        if (balance <= commission_in_apt) {
+            // If balance is exactly equal to commission in APT, this will set commission_debt to 0.
+            let debt_apt = commission_in_apt - balance;
+            config.commission_debt = (math128::mul_div((debt_apt as u128), oracle::precision(), apt_price) as u64)
+        } else {
+            let surplus_balance = balance - commission_in_apt;
+            aptos_account::transfer(commission_signer, config.manager, surplus_balance);
+        };
```

Comment:
Change the debt to zero here

Reply:
It's already updated outside of the function for less branching